provider/pkg/client/client.go

```diff
@@ -64,41 +64,34 @@ func (c *clientImpl) Create(ctx context.Context, typeName string, desiredState m
 	}
 
 	pi, waitErr := c.awaiter.WaitForResourceOpCompletion(ctx, res)
-
-	// Read the state - even if there was a creation error but the progress event contains a resource ID.
-	var id string
-	var readErr error
-	if pi != nil && pi.Identifier != nil {
-		// Retrieve the resource state from AWS.
-		// Note that we do so even if creation hasn't succeeded but the identifier is assigned.
-		id = *pi.Identifier
-		resourceState, err = c.api.GetResource(ctx, typeName, id)
-		if err != nil {
-			readErr = fmt.Errorf("reading resource state: %w", err)
-		}
-	}
-
 	if waitErr != nil {
-		if id == "" {
-			return nil, nil, waitErr
+		if pi == nil || pi.Identifier == nil {
+			return nil, nil, fmt.Errorf("creating resource (await): %w", waitErr)
 		}
-
-		if readErr != nil {
-			return nil, nil, fmt.Errorf("resource partially created but read failed. read error: %v, create error: %w", readErr, waitErr)
+		if pi.ErrorCode == types.HandlerErrorCodeAlreadyExists {
+			// Already Exists is a special case because it's the scenario when we can't proceed to resource read
+			// to validate its existence. We should return immediately as a hard error.
+			return nil, nil, waitErr
 		}
-
-		// Resource was created but failed to fully initialize.
-		// If it has some state, return a partial error.
-		return &id, resourceState, waitErr
 	}
 	if pi.Identifier == nil {
 		return nil, nil, errors.New("received nil identifier while reading resource state")
```

Comment:
nit. this error message doesn't quite make sense? We're not really in the middle of reading; we can't read b/c there was no id returned from the await